docs/testplan/GCU-Dynamic-ACL-testplan.md

```diff
@@ -0,0 +1,464 @@
+                "DYNAMIC_ACL_TABLE_TYPE" : {
+                "MATCHES": ["DST_IP","DST_IPV6","IN_PORTS"],
+                "ACTIONS": ["PACKET_ACTION","COUNTER"],
+                "BIND_POINTS": ["PORT"]
+                }
+            }
+        }
+    ]
+**Expected Result**
+- Operation Success
+
+**Additional checks**
+- None
+
+### Create an ACL Table
+
+**Json Patch**:
+
+    [
+        {
+            "op": "add",
+            "path": "/ACL_TABLE/DYNAMIC_ACL_TABLE",
+            "value": {
+                "policy_desc": "DYNAMIC_ACL_TABLE",
+                "type": "DYNAMIC_ACL_TABLE_TYPE",
+                "stage": "INGRESS",
+                "ports": {vlan port members from minigraph}
+            }
+        }
+    ]
+
+**Expected result**
+- Operation Success
+
+
+**Additional checks**
+- Check that results of the command “show acl table” match this expected output:
+
+Name  | Type | Binding | Description | Stage | Status
+------------- | ------------- | ---------- | ----------| --------- | ---------
+DYNAMIC_ACL_TABLE | DYNAMIC_ACL_TABLE_TYPE | {vlan port 1} | DYNAMIC_ACL_TABLE_TYPE | ingress | Active
+  | | {vlan port 2} | |
+  | | {vlan port 3}...
+
+### Create Forwarding Rules
+
+**Json Patch:**
+
+    [
+        {
+            "op": "add",
+            "path": "/ACL_RULE",
+            "value": {
+                "DYNAMIC_ACL_TABLE|RULE_1": {
+                    "DST_IP": "103.23.2.1/32",
+                    "PRIORITY": "9999",
+                    "PACKET_ACTION": "FORWARD"
+                },
+                "DYNAMIC_ACL_TABLE|RULE_2": {
+                    "DST_IPV6": "103:23:2:1::1/128",
+                    "PRIORITY": "9998",
+                    "PACKET_ACTION": "FORWARD"
+                }
+            }
+        }
+    ]
+
+**Expected Result**
+- Operation Success
+
+**Additional Checks**
++ Check that results of “show acl rule DYNAMIC_ACL_TABLE RULE_1” and “show acl rule DYNAMIC_ACL_TABLE RULE_2” match the following output:
+  + DYNAMIC_ACL |_TABLE	RULE_1 | 9999 | FORWARD |  DST_IP: 103.23.2.1/32
+  + DYNAMIC_ACL_TABLE | RULE_2 | 9998 | FORWARD | DST_IPV6: 103.23.2.1::1/128
+
+### Create Drop Rule
+
+**Json Patch**
+
+    [
+        {
+            "op": "add",
+            "path": "/ACL_RULE/DYNAMIC_ACL_TABLE|RULE_3",
+            "value": {
+                "PRIORITY": "9997",
+                "PACKET_ACTION": "DROP",
+                "IN_PORTS": {port selected from DUT minigraph}
+            }
+        }
+    ]
+
+    OR
+
+    [
+        {
+            "op": "add",
+            "path": "/ACL_RULE",
+            "value": {
+                "DYNAMIC_ACL_TABLE|RULE_3": {
+                    "PRIORITY": "9997",
+                    "PACKET_ACTION": "DROP",
+                    "IN_PORTS": setup["blocked_src_port_name"],
+                }
+            }
+        }
+    ]
+
+    Which patch is applied depends on whether there are already ACL Rules created, or if this is the first ACL rule that we are creating
```

Comment:
>there are already ACL Rules created, or if this is the first ACL rule that we are creating [](http://example.com/codeflow?start=46&length=90)

Thanks for considering both cases!
By RFC ([https://datatracker.ietf.org/doc/html/rfc6902#section-4.1](https://datatracker.ietf.org/doc/html/rfc6902#section-4.1)), I believe the first patch will always works, could you check?

Reply:
Similarly to what I said in the above comment, the first patch actually errors out if there is not a single ACL rule created.  I observed this behavior on both a KVM and numerous STARLAB testbed.  It was necessary for me to create two different patches - one for when it is added as the initial rule, and one for when it is added as a secondary rule.  Similar behavior is seen when removing rules.  Eventually, it is necessary to remove "path: /ACL_RULE" in its entirety, as attempting to remove the final rule results in an error about an empty table